src/cdomains/addressDomain.ml

```diff
@@ -103,7 +103,86 @@ struct
 
   (* strings *)
   let from_string x = singleton (Addr.from_string x)
+      
   let to_string x = List.filter_map Addr.to_string (elements x)
+      
+  let to_string_length x =
+    let transform elem =
+      match Addr.to_string_length elem with
+      | Some x -> Idx.of_int IUInt (Z.of_int x)
+      | None -> Idx.top_of IUInt in 
```

Comment:
Isn't this assuming `size_t` is `unsigned int`? `Cil.kindOfSizeOf` would probably be better.

Reply:
In *cil.ml*, `Cil.kindOfSizeOf` is defined as `ref IUInt`, so would this make any difference? There is also a signature mismatch problem because of the `ref` if using `Cil.kindOfSizeOf`